test/integration/http_integration.h

```diff
@@ -366,8 +366,6 @@ class HttpIntegrationTest : public BaseIntegrationTest {
   testing::NiceMock<Random::MockRandomGenerator> random_;
   Quic::QuicStatNames quic_stat_names_;
   std::string san_to_match_{"spiffe://lyft.com/backend-team"};
-  bool enable_quic_early_data_{true};
-  std::vector<absl::string_view> custom_alpns_;
```

Comment:
I had to refactor the tests so that I could modify the behaviors with a config modifier, so this was moved elsewhere.

Reply:
Can you say more about the need to move the transport socket config change from initializer() to constructor? It actually removes the ability of this test setup to pass in a YAML config with customized QUIC transport socket factory because the transport socket factory gets overridden again in the constructor? I guess this is why you have to move early data and custom ALPNs into config modifier. But if possible, it will be better to still allow a subclass of HttpIntegrationTest to take a different YAML config and respect whatever is specified in the YAML.

Reply:
It was so that I could add this modifer: https://github.com/envoyproxy/envoy/pull/36926/files#diff-74df412a442c8c371207c62ffba85f3cc584fb6237b09bc07e1e691359e5013cR98

I needed the transport socket configuration to be setup before the config_helper_ modifiers get run, otherwise the later configuration of the transport socket would overwrite my earlier modification.

I have no objection to further refactors or changes to how these tests work to allow what you want/need here; I did the refactoring in this PR to cover what I needed, and it seemed to support all previously existing test cases.